crates/bevy_ecs/src/system/commands/error_handler.rs

```diff
@@ -1,32 +1,53 @@
 //! This module contains convenience functions that return simple error handlers
-//! for use with the following methods:
-//! - [`Commands::queue_fallible_with`](super::Commands::queue_fallible_with)
-//! - [`Commands::override_error_handler`](super::Commands::override_error_handler)
-//! - [`EntityCommands::queue_with`](super::EntityCommands::queue_with)
-//! - [`EntityCommands::override_error_handler`](super::EntityCommands::override_error_handler)
-//! - [`EntityEntryCommands::override_error_handler`](super::EntityEntryCommands::override_error_handler)
+//! for use with [`Commands::queue_handled`](super::Commands::queue_handled) and [`EntityCommands::queue_handled`](super::EntityCommands::queue_handled).
 
+use crate::{result::Error, world::World};
 use log::{error, warn};
 
-use crate::{system::CommandError, world::World};
-
 /// An error handler that does nothing.
-pub fn silent() -> fn(&mut World, CommandError) {
+pub fn silent() -> fn(&mut World, Error) {
     |_, _| {}
 }
 
 /// An error handler that accepts an error and logs it with [`warn!`].
-pub fn warn() -> fn(&mut World, CommandError) {
+pub fn warn() -> fn(&mut World, Error) {
     |_, error| warn!("{error}")
 }
 
 /// An error handler that accepts an error and logs it with [`error!`].
-pub fn error() -> fn(&mut World, CommandError) {
+pub fn error() -> fn(&mut World, Error) {
     |_, error| error!("{error}")
 }
 
 /// An error handler that accepts an error and panics with the error in
 /// the panic message.
-pub fn panic() -> fn(&mut World, CommandError) {
+pub fn panic() -> fn(&mut World, Error) {
     |_, error| panic!("{error}")
 }
+
+/// The default error handler. This defaults to [`panic()`]. If the
+/// `configurable_error_handler` cargo feature is enabled, then
+/// `GLOBAL_ERROR_HANDLER` will be used instead, enabling error handler customization.
+#[cfg(not(feature = "configurable_error_handler"))]
+#[inline]
+pub fn default() -> fn(&mut World, Error) {
+    panic()
+}
+
+/// A global error handler. This can be set at startup, as long as it is set before
+/// any uses. This should generally be configured _before_ initializing the app.
+///
+/// If the `configurable_error_handler` cargo feature is enabled, this will be used
```

Comment:
OnceLocks with a function pointer are a bit exotic for something that's this user facing. Can we add a simple doc test teaching users how to set this?